src/main/java/hu/blackbelt/structured/map/proxy/MapProxy.java

```diff
@@ -706,6 +708,80 @@ private boolean invokeEquals(Object proxy, Object[] args) throws InvocationTarge
         return false;
     }
 
+    private void invokeAdd(Method m, Object[] args) throws ExecutionException {
+        if (params.isImmutable()) {
+            throw new IllegalStateException("Could not call set on immutable object");
+        }
+        String attrName = Character.toLowerCase(m.getName().charAt(5)) + m.getName().substring(6);
+        AttributeInfo attributeInfo = typeInfoCache.get(clazz).get(attrName);
+        Object values = internal.get(getKeyName(clazz, attrName));
+
+        List<Object> valuesArray = (values instanceof Collection<?>)
+                ? valuesArray = new ArrayList<>((Collection) values)
+                : new ArrayList<>();
+
+        if (args[0] instanceof Object[]) {
+            Object[] valuesForAdd = (Object[]) args[0];
+            for (Object value : valuesForAdd) {
+                if (attributeInfo != null && attributeInfo.isComposite() && value instanceof MapHolder) {
+                   if (attributeInfo.getPropertyType().isInterface()) {
+                       MapHolder proxy = (MapHolder) MapProxy.builder(attributeInfo.getPropertyType())
+                               .withParams(params)
+                               .withMap(((MapHolder) value).toMap())
+                               .newInstance();
+                       proxy.toMap().entrySet().forEach(e -> {
+                           internal.put(e.getKey(), e.getValue());
+                       });
+                   } else {
```

Comment:
This part can be extracted I think, because it is used in `invokeSet` and a variation in `invokeRemove` too.

Reply:
+1

Reply:
[b7d7975](https://github.com/BlackBeltTechnology/structured-map-proxy/pull/82/commits/b7d79756f1b886512710a06d2298613ff0d17eae)